lib/mongo/server/connection_common.rb

```diff
@@ -52,6 +52,38 @@ def connected?
       # @api private
       attr_reader :pid
 
+      # Build a document that should be used for connection handshake.
+      #
+      # @param [Server::AppMetadata] app_metadata Application metadata
+      #
+      # @return [BSON::Document] Document that should be sent to a server
+      #     for handshake purposes.
+      #
+      # @api private
+      def handshake_document(app_metadata)
+        document = if app_metadata.server_api && app_metadata.server_api[:version]
+                     HELLO_OP_QUERY
```

Comment:
This is going to get replaced by hello op message later, right? If yes can you add a brief comment saying so please.

Reply:
Comment added close to places where we build messages.